### ISSUE TITLE
Success Criterion 3.2.4 - Consistent Identification - Level AA

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#55]
WCAG2ICT guidance: https://www.w3.org/TR/wcag2ict-22/#consistent-identification

Share your thoughts for applying to mobile apps as a comment below.

**Comment by Keanem6** [comment]
Once a definition is created for 'views' then i can see great value in applying this.

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>16 October 2024</summary>

[Source](https://www.w3.org/2024/10/16-matf-minutes.html#caab)

~~~text
JJ - WCAG2ICT replaces "set of web pages" with "a software program"

JJ - correction - EN 301 549 replaces "set of web pages" with "a software program"

JJ - This mostly related to how we replace "set of web pages".

<JJ> Mick: in our group we would look at a relatively small set of components for consistent navigations, such as search buttons?

JJ - I would consider a back button, close button, buttons in a list, a change of tab in a tab bar.

Aash - should we introduce certain things - the bottom sheets without the close button. Very hard to dismiss those overlays.

JJ - I think it would be interesting to add such things in a Note. That might be more related to Keyboard Operation rather than this success criterion

JJ - I wonder would a back button or close button placed differently would fail this criteria in app or is it too strict. Is it more about large lists of links rather than a single link/button.

JJ - Does 'navigational mechanisms' related to one button or link

Jamie - this criteria isn't about where the back button is on every app

ACTION: Define "navigational mechanisms"

Jamie - what are we considering navigational mechanisms? It would be helpful outlining examples.

JJ - this is specific to a single app.
~~~
</details>

**Comment by JJdeGroot** [comment]
**[Text in EN 301 549 v4.1.1](https://www.etsi.org/deliver/etsi_en/301500_301599/301549/04.01.01_60/en_301549v040101p.pdf#page=99)**

> 11.3.2.4 Consistent identification
> 
> Where ICT is, or includes, non-web software that provides a user interface,
> components that have the same functionality within the non-web software program shall be identified consistently,
> except where not doing so is essential to the function of the non-web software.
> 
> **NOTE 1:** Consistent does not mean identical, but that the characteristics of the identifiers (such as text and look)
> are as similar as practical to reflect the fact that they identify the same functionality.
> 
> **NOTE 2:** This requirement is based on WCAG 2.2 3.2.4 "Consistent identification", but replacing the scope from a
> "set of web pages" to "a non-web software program", in order to make it more relevant for non-web
> software.

**[Text in WCAG2ICT Editor's Draft (18 September 2026)](https://w3c.github.io/wcag2ict/#consistent-identification)**

> This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.2.4](https://www.w3.org/WAI/WCAG22/Understanding/consistent-identification#intent), replacing “set of web pages” with “set of non-web documents" or "set of non-web software programs”.
> 
> With these substitutions, it would read:
> 
> 3.2.4 Consistent Identification: Components that have the [same functionality](https://w3c.github.io/wcag2ict/#dfn-same-functionality) within a [[set of non-web documents](https://w3c.github.io/wcag2ict/#set-of-documents)] or a [[set of non-web software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs)] are identified consistently.
>
> **Note 1 (Added)**
> 
> See [set of documents](https://w3c.github.io/wcag2ict/#set-of-documents) and [set of software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs) in the Key Terms section to determine when a group of documents or software programs is considered a set for this success criterion. Those implementing this document (WCAG2ICT) will need to consider if this success criterion is appropriate to apply to non-web documents and non-web software. See the [Interpretation of Web Terminology in a Non-web Context](https://w3c.github.io/wcag2ict/#interpretation-of-web-terminology-in-a-non-web-context).
> 
> **Note 2 (Added)**
> 
> Although not required by this success criterion, ensuring that component identification be consistent when they occur more than once within non-web documents or non-web software programs directly addresses user needs identified in the Intent section for this success criterion, and is generally considered best practice.
> 
> **Note 3 (Added) (for non-web software)**
> 
> Sets of software that meet this definition appear to be extremely rare.
> 
> **Note 4 (Added) (for non-web software)**
> 
> For general guidance, see also the [comments on closed functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality). For commentary on the individual success criterion, see [Success criteria problematic for closed functionality, Consistent Identification](https://w3c.github.io/wcag2ict/#problematic-for-closed-consistent-identification).
